pkg/proxy/proxy.go

```diff
@@ -165,6 +172,16 @@ func proxyTypeNotFoundError(proxyType types.ProxyType, listener string, ingress
 	return fmt.Errorf("unrecognized %s proxy type for %s: %s", dir, listener, proxyType)
 }
 
+func (p *Proxy) UpdateSDP(rules map[identity.NumericIdentity]policy.SelectorPolicy) {
+	if p.policyUpdater == nil {
+		return
+	}
+
+	p.mutex.Lock()
```

Comment:
why lock?

Reply:
Proxy while sending data to embedded dns proxy takes a lock. I want to make sure that we do the same while sending data to sdp.